docs/type_inference.md

```diff
@@ -0,0 +1,60 @@
+# Type Inference
+
+StableHLO is an operation set that expresses ML computations. It has been originally bootstrapped from [the MHLO dialect](https://github.com/tensorflow/mlir-hlo#meta-hlo-dialect-mhlo), including inheriting the implementation of type inference. The implementation progress is tracked in [status.md](https://github.com/openxla/stablehlo/blob/main/docs/status.md). 
+
+To implement high-quality verifiers and shape functions for StableHLO ops, these guidelines and criterion are proposed below to follow:
```

Comment:
Drop "and criterion"? Or reformulate as "and criteria".

Reply:
Removed.